components/builder-api/src/server/resources/channels.rs

```diff
@@ -31,354 +33,314 @@ impl Channels {
     //
     // Internal - these functions should return Result<..>
     //
+    fn do_promote_package(
+        req: &HttpRequest<AppState>,
+        origin: String,
+        channel: String,
+    ) -> Result<NetOk> {
+        let ident = OriginPackageIdent::new();
+
+        if !helpers::check_origin_access(req, &origin).is_err() {
+            return Err(Error::NetError(NetError::new(
+                ErrCode::ACCESS_DENIED,
+                "core:promote-package-to-channel:0",
+            )));
+        }
 
-    //
-    // Route handlers - these functions should return HttpResponse
-    //
+        let mut origin_get = OriginGet::new();
+        origin_get.set_name(ident.get_origin().to_string());
 
-    //
-    // Route registration
-    //
-    pub fn register(app: App<AppState>) -> App<AppState> {
-        app
-    }
-}
+        let origin_id = match route_message::<OriginGet, Origin>(req, &origin_get) {
```

Comment:
Mjght be a good time to look whether we need all these Channel.. errors - I think we can just remove channel specific errors as they are not getting used upstream ever... so this paragraph essentially becomes:
```
let origin_id = route_message::<OriginGet, Origin>(req, &origin_get)?;
```